spec/requests/api/v1/registrations/create_spec.rb

```diff
@@ -0,0 +1,80 @@
+require 'rails_helper'
+
+RSpec.describe 'POST /api/v1/users', type: :request do
+  subject { post user_registration_url, params: user_params, as: :json }
+
+  context 'when the request is valid' do
+    let(:user_params) do
+      {
+        user: attributes_for(:user)
+      }
+    end
+
+    it 'returns the user' do
+      subject
+      expect(json).to include_json(
+        user: {
+          email: user_params[:user][:email],
+          provider: user_params[:user][:provider],
+          full_name: user_params[:user][:full_name],
+          gender: user_params[:user][:gender],
+          uid: user_params[:user][:uid]
+        }
+      )
+    end
+
+    it 'returns a success status' do
+      subject
+      expect(response).to be_successful
+    end
+
+    it 'creates a user' do
+      expect { subject }.to change { User.count }.from(0).to(1)
+    end
+  end
+
+  context 'when user email is already registrated' do
+    let!(:user) { create(:user) }
+    let(:user_params) do
+      {
+        user: attributes_for(:user).except(:email).merge(email: user.email)
```

Comment:
Here you can do `attributes_for(:user, email: user.email)` instead of having the `except` and `merge`